### PR TITLE
Updating naming in new backend

### LINKED ISSUE #23215: ISIS SANS update workspace nameing to match v1
The names of the default output workspaces should match that currently happening in version one.

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**

This updates the naming in the new GUI to match scientific specifications. After consulting the scientists the following naming scheme was decided upon:
* If no user specified name is provided the name should consist of {RunNumber}{Detector}_{Dimensionality}_{WavelengthRange}{PhiRange}_{TimeSlice}
* If a user specified name is provided the output workspaces are given this name. Suffixes are appended in the case of merged, all, or sliced reductions in order to distinguish the different workspaces.
* Also as part of this PR the input data is grouped and the transmission workspaces are grouped.
* The wavelength range over which the transition workspace is fitted is also adjusted. 

**To test:**
A good place to start testing would be:
* Get the data from [loqdemo.zip](https://github.com/mantidproject/mantid/files/2286736/loqdemo.zip) and insure the directory is added to the Mantid search paths.
* Then run the following script
```
from sans.command_interface.ISISCommandInterface import *

MASKFILE = FileFinder.getFullPath('MaskFile.txt')
BATCHFILE = FileFinder.getFullPath('batch_mode_reduction.csv')
Clean()
LOQ()
MaskFile(MASKFILE)
SetEventSlices('3:2:9')
BatchReduce(BATCHFILE, format='.nxs', saveAlgs={"SaveCanSAS1D": ".xml"}, combineDet='merged')
```

Check that 5 workspace groups are created once the script has finished running. Check that these make sense and hopefully look like the image below.

image

* You can modify the batch file in order to test in the case of a non user specified name. Simply open it in excel or similar and remove the entry entitled 'first_time'

Fixes #23215 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/test/SANS/common/general_functions_test.py [hunk 1]
```diff
@@ -1,18 +1,25 @@
...
 from sans.common.constants import (SANS2D, LOQ, LARMOR)
 from sans.common.enums import (ISISReductionMode, ReductionDimensionality, OutputParts,
-                               SANSInstrument, DetectorType, SANSFacility)
+                               SANSInstrument, DetectorType, SANSFacility, DataType)
 from sans.test_helper.test_director import TestDirector
 from sans.state.data import StateData
+from sans.state.save import get_save_builder
+
+if sys.version_info.major > 2:
+    from unittest import mock
```
Comment (reimundILL) [inline scripts/test/SANS/common/general_functions_test.py]: Can you point me to where this import is used? Mocking seems to be a creat tool here.

Comment (Mantid-Matthew) [reply scripts/test/SANS/common/general_functions_test.py]: Looks like I added the import but the never used it in this case. I'll remove it now.

**Review by soininen — CHANGES_REQUESTED** [review]
The test script provided in the description creates the expected groups and workspaces, although there were some `Workspace already exists in a WorkspaceGroup` warnings. Can these be ignored? The code looks good to me and numerous unit tests have been added :+1: Was there some reason the release notes were not updated here?

File: scripts/test/SANS/common/general_functions_test.py [hunk 2]
```diff
@@ -1,18 +1,20 @@
 from __future__ import (absolute_import, division, print_function)
 import unittest
+import sys
```
Comment (reimundILL) [inline scripts/test/SANS/common/general_functions_test.py]: Since no mocking, can this import be removed?

**Comment by Mantid-Matthew** [comment]
I've now added a check of whether a workspace is already in the group which should get rid off the warning messages. I've also added to the release notes and removed the unneeded imports from the test.

**Review by soininen — APPROVED** [review]

**Comment by soininen** [comment]
The failing RHEL7 test seems to be unrelated.

**Pull request merged**